spec/classes/config/global/auth_spec.rb

```diff
@@ -68,6 +78,72 @@
       expect(content).to include('testuser:password,user,deploy')
       expect(content).to include('anotheruser:anotherpassword,user')
     end
+
+    it 'should contain PropertyFileLoginModule and be required' do
+      jaas_auth = catalogue.resource('file', '/etc/rundeck/jaas-auth.conf')[:content]
+      expect(jaas_auth).to include('org.eclipse.jetty.plus.jaas.spi.PropertyFileLoginModule required')
+    end
+  end
+
+  describe 'with multiauth ldap and file auth users array' do
+    let(:params) do
+      {
+        :auth_types  => %w(ldap file),
+        :auth_config => {
+          'file' => {
+            'auth_users' => [
+              {
+                'username' => 'testuser',
+                'password' => 'password',
+                'roles'    => %w(user deploy),
+              },
+              {
+                'username' => 'anotheruser',
+                'password' => 'anotherpassword',
+                'roles'    => ['user'],
+              },
+            ],
+          },
+
+          'ldap' => {
+            'debug'                   => 'true',
+            'url'                     => 'localhost:389',
+            'force_binding'           => 'true',
+            'force_binding_use_root'  => 'true',
+            'bind_dn'                 => 'test_rundeck',
+            'bind_password'           => 'abc123',
+            'user_base_dn'            => 'ou=users,ou=accounts,ou=corp,dc=xyz,dc=com',
+            'user_rdn_attribute'      => 'sAMAccountName',
+            'user_id_attribute'       => 'sAMAccountName',
+            'user_password_attribute' => 'unicodePwd',
+            'user_object_class'       => 'user',
+            'role_base_dn'            => 'ou=role based,ou=security,ou=groups,ou=test,dc=xyz,dc=com',
+            'role_name_attribute'     => 'cn',
+            'role_member_attribute'   => 'member',
+            'role_object_class'       => 'group',
+            'supplemental_roles'      => 'user',
+            'nested_groups'           => 'true',
+          },
+        },
+      }
+    end
+
+    it 'should generate valid content for realm.properties' do
+      content = catalogue.resource('file', '/etc/rundeck/realm.properties')[:content]
+      expect(content).to include('admin:admin,user,admin,architect,deploy,build')
+      expect(content).to include('testuser:password,user,deploy')
+      expect(content).to include('anotheruser:anotherpassword,user')
+    end
+
+    it 'should contain PropertyFileLoginModule and be required' do
+      jaas_auth = catalogue.resource('file', '/etc/rundeck/jaas-auth.conf')[:content]
+      expect(jaas_auth).to include('org.eclipse.jetty.plus.jaas.spi.PropertyFileLoginModule required')
+    end
+
+    it 'should contain JettyCachingLdapLoginModule and be required' do
```

Comment:
Perhaps the title should read `should contain JettyCachingLdapLoginModule and be sufficient` given that is what is being tested.